src/index.ts

```diff
@@ -11,29 +11,30 @@ import {
   GraphQLOutputType,
   GraphQLScalarType,
   GraphQLSchema,
-  GraphQLType,
 } from "graphql";
-import { PluginFunction, Types } from "@graphql-codegen/plugin-helpers";
+import { Code, code, imp } from "ts-poet";
 import { SymbolSpec } from "ts-poet/build/SymbolSpecs";
 import PluginOutput = Types.PluginOutput;
 
 /** Generates `newProject({ ... })` factory functions in our `graphql-types` codegen output. */
 export const plugin: PluginFunction = async (schema, documents, config: Config) => {
   const chunks: Code[] = [];
 
-  // Establish defaults for interfaces, i.e. Named --> assume its Author
-  const interfaceDefaults: Record<string, string> = {};
+  // Create a map of interface -> implementing types
+  const interfaceImpls: Record<string, string[]> = {};
   Object.values(schema.getTypeMap()).forEach(type => {
     if (type instanceof GraphQLObjectType) {
       for (const i of type.getInterfaces()) {
-        if (interfaceDefaults[i.name] === undefined) {
-          interfaceDefaults[i.name] = type.name;
+        if (interfaceImpls[i.name] === undefined) {
+          interfaceImpls[i.name] = [];
         }
+        interfaceImpls[i.name].push(type.name);
       }
     }
   });
```

Comment:
FYI @stephenh - I pulled this out of the `possible-types` plugin. There is only about 5 lines of `unique code` in that plugin. Maybe we should just move it in here?

Reply:
Hm, yeah, I go back & forth. It seems like we are definitely getting to have a lot of plugins. And it hurts the eventual goal of having a "minimal, out-of-the-box, everything just works setup". But kinda wondering if we keep the plugins ruthlessly separate, and then end up having like a `@homebound/graphql-client-plugins` or something that is a meta plugin that combines all of them to just work. Or something.